maintenance/wikia/WikiaMaps/removeImapTags.php

```diff
@@ -0,0 +1,223 @@
+ * @usage SERVER_ID=203236 php removeImapTags.php --tiles-set-id=1 --quiet --test
+ */
+
+require_once( __DIR__ . '/../../Maintenance.php' );
+
+ini_set( "include_path", __DIR__ . "/../../" );
+ini_set('display_errors', 'stderr');
+ini_set('error_reporting', E_ALL);
+
+class RemoveImapTags extends Maintenance {
+	static protected $test = false;
+
+	private $maps;
+
+	/**
+	 * @var Array
+	 */
+	private $mapsClientConfig;
+
+	public function __construct() {
+		parent::__construct();
+		$this->mDescription = "Removes from articles <imap /> tags for given tiles' set id.";
+		$this->addOption( 'test', 'Test mode; make no changes', $required = false, $withArg = false, 't' );
+		$this->addOption( 'maps-db-host', 'WikiaMaps DB host', $required = true, $withArg = true );
+		$this->addOption( 'maps-db-user', 'WikiaMaps DB user', $required = true, $withArg = true );
+		$this->addOption( 'maps-db-pass', 'WikiaMaps DB pass', $required = true, $withArg = true );
+		$this->addOption( 'maps-db-name', 'WikiaMaps DB name', $required = true, $withArg = true );
+		$this->addOption(
+			'tiles-set-id',
+			'Specify a tiles\' set id which leads to a group of maps which <imap /> tags should get removed',
+			$required = true,
+			$withArg = true,
+			'tsi'
+		);
+	}
+
+	static public function isTest() {
+		return self::$test;
+	}
+
+	static public function isNaturalNumber( $int ) {
+		return intval( $int ) > 0;
+	}
+
+	public function getMapsIdsUsingTileset( $cityId, $tilesSetId ) {
+		$dbw = new DatabaseMysqli(
+			$this->getOption('maps-db-host'),
+			$this->getOption('maps-db-user'),
+			$this->getOption('maps-db-pass'),
+			$this->getOption('maps-db-name')
+		);
+
+		return $dbw->selectFieldValues( 'map', 'id', [
+				'city_id' => $cityId,
+				'tile_set_id' => $tilesSetId
+		] );
+	}
+
+	public function getArticlesIdsUsingImap( $cityId ) {
+		global $wgStatsDB;
+
+		$dbw = wfGetDB(DB_SLAVE, array(), $wgStatsDB);
+		return $dbw->selectFieldValues( 'city_used_tags', 'ct_page_id', [
+			'ct_wikia_id' => $cityId,
+			'ct_kind' => 'imap'
+		] );
+	}
+
+	public function hasImapTag( $articleId, &$foundTags, &$foundTagsMapIds ) {
+		$article = Article::newFromID( $articleId );
+
+		if ( $article instanceof Article && $article->getID() ) {
+			$results = null;
+			$articleContent = $article->getContent();
+			$imapSearchRegexp = "/<imap.*map\-id\=['\"](\d{1,})['\"].*(<\/imap|\/)>/";
+
+			$noOfFoundTags = preg_match_all($imapSearchRegexp, $articleContent, $results);
+
+			if( $noOfFoundTags === 0 ) {
+				return false;
+			}
+
+			$foundTags = $results[0];
+			$foundTagsMapIds = $results[1];
+
+			return true;
+		} else {
+			$this->output( sprintf( 'Article #%d does not exist anymore', $articleId ) . PHP_EOL );
+		}
+	}
+
+	public function hasTagsToRemove( $foundTagsMapIds, $mapsUsingTheTileset, &$tagsToRemoveArrayKeyIds ) {
+		foreach( $foundTagsMapIds as $key => $mapId ) {
+			if ( in_array( $mapId, $mapsUsingTheTileset ) ) {
+				$tagsToRemoveArrayKeyIds[] = $key;
```

Comment:
Can we leave earlier and just call `return true` here and `return false` after foreach loop?